src/server.rs

```diff
@@ -417,26 +417,46 @@ mod test {
 
     /// Attempts to grab a local, unbound socket address for testing.
     fn get_unbound_address() -> SocketAddr {
-        TcpListener::bind("127.0.0.1:0").unwrap().local_addr().unwrap()
+        let mut current = 4001;
```

Comment:
Hm, perhaps I misunderstood the goal of the function? I thought it was intended to get a free address.